### DISCUSSION TITLE
using the pdf on the web

### DISCUSSION THREAD

**Discussion by ghost** [pdf-association/pdf-issues#233] [Q&A] [answered]
Hey there!

#### Initial problem
I was wondering if there is any way for the contents of a PDF to be updated after being distributed. 

#### Possible solution
To make this idea possible, an editable web-pdf file format is needed.

#### Reasons
1. If the PDF distribution is editable and offline, then people can edit PDF without needing text editors installed on their machines, being able to edit the PDF document itself directly in the browser.
2. This can be useful in case of academic research, where you could download the pdf of some field of knowledge and edit that pdf without any text editor, that is, editing offline the pdf through the browser itself. 
3. An initial advantage would be to use more pdf than proprietary formats like doc, docx etc. 

#### Possible implementation
I thought of the tiddlywiki format which works offline and is editable in the browser - but for pdf. 

#### Doubt
1. Is it possible or would it be possible to have a pdf file format on the web?
2. Have you all ever thought about implementing some idea or concept of a web editable pdf file?
3. I wonder if there is any possibility to turn pdf files into tiddlywiki, and tiddlywiki files into pdf files. It is possible? 

*Not a feature request, but a technical question.*

**Comment by petervwyatt** [comment, marked as answer]
PDF the file format can already support this, but the solution invokes both security concerns and goes against a fundamental principle of PDF being a reliable "document of record". And I would suspect very very few, if any, mainstream PDF viewers will support this (not the least because viewers often render and cache pages).

Any PDF stream object can have **F** and **FFilter** entries, as per Table 5 in ISO 32000-2. **F** is a file specification dictionary that _shall be_ used in place of any data in the PDF between the **stream** and **endstream** keywords. And a file specification dictionary can be a reference to a URL by setting the **FS** key to _URL_ and setting the value of the **F** key to the URL - see subclause 7.11.5 URL specification which has an example. So if you set the content stream of an XObject to be such a URL-based stream then theoretically this would achieve what you want. 

Note that PDF the file format does not define _when_ the URL is fetched so there is **NO** requirement that a processor needs to fetch the URL each time the page or XObject is displayed, scrolled into view, live updated/polled, etc. The minimum expectation is that the URLs necessary to show pages would be fetched once when the PDF is first opened (each time).

**Reply by ghost** [reply]
Hi petervwyatt, thank you for feedback.
